From 668f61ac543939d2a1e27c0c80514485552027b7 Mon Sep 17 00:00:00 2001
From: Fabian Schramm <55981657+fabinsch@users.noreply.github.com>
Date: Tue, 15 Nov 2022 17:06:55 +0100
Subject: [PATCH] fix python interface

---
 interfaces/python/setup.py | 9 ++++++---
 1 file changed, 6 insertions(+), 3 deletions(-)

diff --git a/interfaces/python/setup.py b/interfaces/python/setup.py
index 913f8da..e0b7b05 100755
--- a/interfaces/python/setup.py
+++ b/interfaces/python/setup.py
@@ -59,12 +59,15 @@ extra_params['library_dirs'] = ['/usr/lib', os.path.join(BASEDIR, 'bin')]
 extra_params['language'] = 'c++'
 
 if platform.system() in ['Linux', 'Darwin']:
-    extra_params['extra_compile_args'] = ['-D__USE_LONG_INTEGERS__',
+    extra_params['extra_compile_args'] = [
+            # '-D__USE_LONG_INTEGERS__',
             '-D__USE_LONG_FINTS__']
 
 if platform.system() == 'Darwin':
+    # extra_params['include_dirs'].append(
+    #         '/Library/Developer/CommandLineTools/usr/include/c++/v1')
     extra_params['include_dirs'].append(
-            '/Library/Developer/CommandLineTools/usr/include/c++/v1')
+            '/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk')
     extra_params['extra_compile_args'] += ['-stdlib=libc++',
         '-Wno-c++11-long-long']
     extra_params['extra_link_args'] = ['-stdlib=libc++'] # override the others!
@@ -89,7 +92,7 @@ if os.name == 'posix':
         "-finline-functions",
         "-fPIC",
         "-DLINUX",
-        "-D__USE_LONG_INTEGERS__",
+        # "-D__USE_LONG_INTEGERS__",
         "-D__USE_LONG_FINTS__",
         "-D__NO_COPYRIGHT__",
     ]
-- 
2.25.1

